tests/unittests/Foundation/NSURLTests.m

```diff
@@ -23,154 +23,169 @@
-                    testURL3,
-                    extension,
-                    [[NSURL alloc] initWithString:@"http://www.test.com/home/asdf/./index.html.txt?Foo#24"]);
-
-    testNSURLMethod(@selector(URLByAppendingPathExtension:),
-                    testURL4,
-                    extension,
-                    [[NSURL alloc] initWithString:@"file://www.test.com/home/../home/./index.html.txt?Foo#24"]);
-
-    testNSURLMethod(@selector(URLByAppendingPathExtension:),
-                    testURL5,
-                    extension,
-                    [[NSURL alloc] initWithString:@"http://www.test.com.txt?Foo#24"]);
-
-    testNSURLMethod(@selector(URLByAppendingPathExtension:),
-                    testURL6,
-                    extension,
-                    [[NSURL alloc] initWithString:@"http://www.test.com/home.page/asdf.txt"]);
-
-    testNSURLMethod(@selector(URLByAppendingPathExtension:),
-                    testURL7,
-                    extension,
-                    [[NSURL alloc] initWithString:@"http://www.test.com/.foo.txt"]);
-
-    [testURL release];
-    [testURL2 release];
-    [testURL3 release];
-    [testURL4 release];
-    [testURL5 release];
-    [testURL6 release];
-    [testURL7 release];
+    TEST_NSURL_METHOD(@selector(URLByAppendingPathExtension:),
+                      testURL,
+                      extension,
+                      [[[NSURL alloc] initWithString:@"http://www.test.com/home/index.html.txt?Foo#24"] autorelease]);
+
+    TEST_NSURL_METHOD(@selector(URLByAppendingPathExtension:),
+                      testURL2,
+                      extension,
+                      [[[NSURL alloc] initWithString:@"http://www.test.com/.txt"] autorelease]);
+
+    TEST_NSURL_METHOD(@selector(URLByAppendingPathExtension:),
+                      testURL3,
+                      extension,
+                      [[[NSURL alloc] initWithString:@"http://www.test.com/home/asdf/./index.html.txt?Foo#24"] autorelease]);
+
+    TEST_NSURL_METHOD(@selector(URLByAppendingPathExtension:),
+                      testURL4,
+                      extension,
+                      [[[NSURL alloc] initWithString:@"file://www.test.com/home/../home/./index.html.txt?Foo#24"] autorelease]);
+
+    TEST_NSURL_METHOD(@selector(URLByAppendingPathExtension:),
+                      testURL5,
+                      extension,
+                      [[[NSURL alloc] initWithString:@"http://www.test.com.txt?Foo#24"] autorelease]);
+
+    TEST_NSURL_METHOD(@selector(URLByAppendingPathExtension:),
+                      testURL6,
+                      extension,
+                      [[[NSURL alloc] initWithString:@"http://www.test.com/home.page/asdf.txt"] autorelease]);
+
+    TEST_NSURL_METHOD(@selector(URLByAppendingPathExtension:),
+                      testURL7,
+                      extension,
+                      [[[NSURL alloc] initWithString:@"http://www.test.com/.foo.txt"] autorelease]);
```

Comment:
No there was a URLByDeletingPathExtension that changed on the reference platform.  I only fixed the memory because I noticed it while removing that one.